apps/lend/src/entities/chain/api/markets-api.ts

```diff
@@ -0,0 +1,17 @@
+import { QueryFunction } from '@tanstack/react-query'
+import { logQuery } from '@/shared/lib/logging'
+import { assertChainValidity } from '../lib/validation'
+import useStore from '@/store/useStore'
+import { ChainQueryKeyType } from '../model'
+
+export const queryOneWayMarketNames: QueryFunction<
+  string[],
+  ChainQueryKeyType<'markets'>
+> = async ({ queryKey }) => {
+  logQuery(queryKey)
+  const [, chainId, , ] = queryKey
+  assertChainValidity({ chainId })
+
+  const {api} = useStore.getState()
+  return api!.oneWayfactory.getMarketList()
```

Comment:
You cannot return undefined from queries. Throw an error if `api` doesn't exists

Reply:
this will not return undefined, it would throw an error if api is undefined. However, the enabled condition checks for the API being in the store